managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java

```diff
@@ -48,6 +53,9 @@ class OpReadEntry implements ReadEntriesCallback {
     public static OpReadEntry create(ManagedCursorImpl cursor, PositionImpl readPositionRef, int count,
             ReadEntriesCallback callback, Object ctx, PositionImpl maxPosition) {
         OpReadEntry op = RECYCLER.get();
+        if (createCount != null) {
```

Comment:
This will introduce a `volatile` read in the prod code which is not free. I'd avoid it if the purpose is for testing.

Reply:
Maybe we can try to recycle the OpReadEntry again in the test to confirm the OpReadEntry is recycled? So that we don't need to introduce this operation.

Reply:
I just removed the `volatile` instead. In non-testing cases, these two variables are always null and never changed and we don't have to guarantee the visibility. Now, the overhead is just a conditional check on normal variables, just like `if (log.isDebugEnabled())` in other places.